File: 04-sorting/sorting/SelectionSort.java
// Selection sort in Java

import java.util.Arrays;

public class SelectionSort {
    static void selectionSort(int[] array) {
        int size = array.length;

        for (int step = 0; step < size - 1; step++) {
            int min_idx = step;

            for (int i = step + 1; i < size; i++) {

                if (array[i] < array[min_idx]) {
                    min_idx = i;
                }
            }

            int temp = array[step];
            array[step] = array[min_idx];
            array[min_idx] = temp;
        }
    }

    public static void main(String args[]) {
        int[] data = { 20, 12, 10, 15, 2 };

        System.out.println("Sorted Array in Ascending Order: ");
        System.out.println(Arrays.toString(data));
    }
}

File: javascript/04-sorting/SelectionSort.js
// Function of selection sort 
const selectionSort = (array)=>{
    for (let i = 0; i < array.length; i++) {
        let minimum_index = i;

        for (let j =i+1;j < array.length;j++) {
            if(array[j]<array[minimum_index]){
                minimum_index=j;
            }

        }
        let temp = array[minimum_index];
        array[minimum_index]=array[i];
        array[i]=temp;
    }
}

let array = [76,57,67,4,9];
selectionSort(array);
console.log(array);